### PR TITLE
[INLONG-8084][DataProxy] Add abnormal event statistics in sink

### LINKED ISSUE #8084: [Improve][DataProxy] Add abnormal event statistics in sink
### Description

At present, the DataProxy node provides abnormal event statistics on the source side and outputs the event to file, but the sink side does not output this content. When the DataProxy node is abnormal, it is troublesome for us to quickly locate and troubleshoot the problem. Through this issue, add an exception event statistics on the sink side.

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gosonzhang**
- Fixes #8084 

1. Summarize the abnormal event statistics function parameters into the common.properties file for configuration;
2. Add abnormal event statistics on the sink side
3. Fix the bug that the proxy and the order-preserving type are not responsed when the message is discarded

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]
LGTM

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source2/BaseSource.java [hunk 1]
```diff
@@ -251,10 +233,14 @@ public synchronized void start() {
                 CommonConfigHolder.getInstance().getClusterName(), getName(), String.valueOf(srcPort));
         MetricRegister.register(metricItemSet);
         // init monitor logic
-        if (fileMetricOn) {
-            this.monitorIndex = new MonitorIndex("Source", monitorStatInvlSec, maxMonitorStatCnt);
+        if (CommonConfigHolder.getInstance().isEnableFileMetric()) {
+            this.monitorIndex = new MonitorIndex("Source",
```
Comment (EMsnap) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source2/BaseSource.java:237]: maybe "Source" should be a constant ?

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/common/SinkContext.java [hunk 2]
```diff
@@ -80,6 +85,16 @@ public SinkContext(String sinkName, Context context, Channel channel) {
      * start
      */
     public void start() {
+        // init monitor logic
+        if (CommonConfigHolder.getInstance().isEnableFileMetric()) {
+            this.monitorIndex = new MonitorIndex("Sink",
```
Comment (EMsnap) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/common/SinkContext.java:90]: I guess Sink should be a constant here

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java [hunk 3]
```diff
@@ -75,6 +75,17 @@ public class CommonConfigHolder {
     // whether enable whitelist, optional field.
     public static final String KEY_ENABLE_WHITELIST = "proxy.enable.whitelist";
     public static final boolean VAL_DEF_ENABLE_WHITELIST = false;
+    // whether enable file metric, optional field.
+    public static final String KEY_ENABLE_FILEMETRIC = "filemetric.enable";
```
Comment (dockerzhang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java:79]: It's better to use `file.metric.enable' to make the configuration easy to understand.

**Review by baomingyu — APPROVED** [review]
+1

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java (hunk 3 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/common/SinkContext.java (hunk 2 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/common/SinkContext.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source2/BaseSource.java (hunk 1 above)
Comment (gosonzhang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source2/BaseSource.java]: done

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**